### DISCUSSION TITLE
Web3Auth Testnet and Dashboard Degredation | Fix by replacing Infura with your own Ropsten RPC

### DISCUSSION THREAD

**Discussion by shahbaz17** [Web3Auth/Web3Auth#685] [Announcements]
Hi 👋 Web3Auth Community,

Rinkeby, Ropsten, and Kovan testnets are deprecated from the Infura API. This has caused our testnet and dashboard to see degredated functionality across our SDKs.

`mainnet/cyan` is not affected by this change, things should be working as expected.

We are working on our side to restore functionality. In the meantime, the hot fix for this is to use another node provider's Ropsten RPC (Alchemy, QuickNode, Chainstack or Ankr) and to swap out infura on our SDKs. Below is an example for the CustomAuth SDK:

Please use your own Quicknode custom URL as `networkUrl`:
`https://{custom-name}.ropsten.quiknode.pro/e2f....5aa679824a`

```js
import TorusSdk from "@toruslabs/customauth";
 
const torusdirectsdk = new TorusSdk({
 baseUrl: `${window.location.origin}/serviceworker`,
 enableLogging: true,
 network: "testnet",
 networkUrl: "https://{custom-name}.ropsten.quiknode.pro/e2f....5aa679824a"
});
```

> Note: It is working with latest version of `"@toruslabs/customauth" -> "11.0.0"`

### If you are not able to see your verifiers on dashboard you can try either of the three solutions given below:-

A. Login on https://dashboard.web3auth.io in a new browser profile.
B. Go to https://app.tor.us and clear the localstorage and login to https://dashboard.web3auth.io
C. Login to https://dashboard.web3auth.io , open the torus wallet from bottom left wallet icon and change the network from ropsten to mainnet from wallet setting and change it back to ropsten.

Either of the above three should be able to show you your old verifiers if not showing now.

**Comment by SurajRtistiq** [comment]
I tried with a QuickNode url but still not working. Please see below code snippet I am using
```
const torusdirectsdk = new TorusSdk({
			baseUrl: `${window.location.origin}/serviceworker`,
			enableLogging: true,
			network: "testnet", // details for test net
			networkUrl: "https://{xxxxx}.ropsten.discover.quiknode.pro/50.............82c"
		});
await torusdirectsdk.init({ skipSw: true });
await torusdirectsdk.getTorusKey("xxxxxx", email, { verifier_id: email }, token);
```

Below is the error
    

>    
>        Error: Verifier not supported. Check if you: 
> 
>       1. Are on the right network (Torus testnet/mainnet) 
> 
>       2. Have setup a verifier on dashboard.web3auth.io?
>     at Torus._callee8$ (torusUtils.esm.js:1288:1)
>     at tryCatch (regeneratorRuntime.js:86:1)
>     at Generator._invoke (regeneratorRuntime.js:66:1)
>     at Generator.next (regeneratorRuntime.js:117:1)
>     at asyncGeneratorStep (asyncToGenerator.js:3:1)
>     at _next (asyncToGenerator.js:25:1) 'mounted caught'

**Comment by MYKEU** [comment]
Still not working for me either using QuickNode - same error as before `Verifier not supported`

**Reply by SurajRtistiq** [reply]
It is working with latest version of "@toruslabs/customauth" -> "11.0.0"

**Reply by himanshuchawla009** [reply]
yes that's correct please update to latest version of android sdk or are you using any other sdk ?

**Reply by lostcodingsomewhere** [reply]
@himanshuchawla009 we are using @toruslabs/customauth 11.0.0 and still experiencing the issue after adding the quick node url

**Comment by himanshuchawla009** [comment]
Hello Guys,
 - If you are using openlogin and its still not working for you just hard reload your app, you don't have to update anything on openlogin.

 - For applications using customAuth mobile SDKs, please update to the latest version of mobile SDKs and pass the `networkUrl` as your own quick node URL for ropsten in the constructor. It won't work in case you are using an older version.

**Reply by Matt561** [reply]
> Hello Guys,
> 
> * If you are using openlogin and its still not working for you just hard reload your app, you don't have to update anything on openlogin.
> * For applications using customAuth mobile SDKs, please update to the latest version of mobile SDKs and pass the `networkUrl` as your own quick node URL for ropsten in the constructor. It won't work in case you are using an older version.

Hard reloading doesn't fix the issue unfortunately. We're using the regular plug and play setup and encountering this randomly. I've tried changing the blockchain that our credentials point to from Ethereum to Solana thinking that it would bypass the infura deprecation issues. We're still unable to use web3auth reliably and it is impacting production environments.

Is there a fix for this in the works and are there any other workarounds you're aware of until a fix is ready?

**Reply by himanshuchawla009** [reply]
Can you check once again plz, it should be working now and this should only affect your testnet site since ropsten was deprecated by infura and we use ropsten in testnet only, there was no issue in mainnet.

I strongly suggest using web3auth mainnet in production, as testnet is always beta.  To verify that you are using mainnet in production check your console logs to see that it should not show you a warning as shown on screenshot below:-

<img width="1116" alt="Screenshot 2022-10-07 at 12 51 34 AM" src="https://user-images.githubusercontent.com/14052374/194372532-c6f37664-5248-4232-b19c-de47b2e8264c.png">

**Reply by FahdW** [reply]
Having same issue, unfortunately testnet remains broken with id token missing

**Reply by himanshuchawla009** [reply]
Hi, plz check now.. idToken should be available now for testnet.

**Comment by agurod42** [comment]
We're using BSC. Should we receive this error? Is there a workaround? We've 60% of our user base waiting for this to be solved.

![console](https://user-images.githubusercontent.com/13837769/194442014-b32be122-68ec-4b7c-b541-2dbba0e09cbb.png)

**Reply by YZhenY** [reply]
just tried the login on your site, it seems to be working with the standard login. what are the steps to replicate?

**Reply by agurod42** [reply]
It's working now. Three hours after my report it started working again. Not sure what happened, but it was on w3a side, for sure. LMK if you need anything on my side that could help you track and solve the issue.

**Comment by hcheng826** [comment]
Hi team,
I am wondering how tKey is related to the RPC url? My understanding is that all the web3auth process: the shares generation, recovery, encryption, decryption all happen off-chain. The only point it involves the chain is when we reconstruct private key and send out the transaction. Is it not the case? And if I am building on chains other than Ropsten, should I set additional configs?
Thanks!

**Comment by yashovardhan** [comment]
Hey @hcheng826,

This annoucement is referring to the web3auth testnet verifiers - not related to any RPC Calls you need to make. Within Web3Auth , to enable the social logins(or tKey service provider), you need to get key shares from our nodes as demonstrated in our [auth network documentation](https://web3auth.io/docs/overview/key-management/#powered-by-the-auth-network). The testnet has dummy nodes set up for testing purposes, which were deployed on ropsten testnet. We were internally using infura for maintaining the uptime of these nodes. However, once they deprecated the testnet, our testnet nodes got down as well. 

You do not need to do any additional changes in terms of using any testnet after the web3auth key generation.

**Comment by joaocosta9** [comment]
Hello guys!
Im working in a project using these deprecated testnets. I was wondering if you have any prediction of how much time it will take to restore this functionality? Just to know if it is worth it to implement a different approach.
Thank you!

**Comment by odanado** [comment]
There seems to have been an announcement about this issue at https://github.com/orgs/Web3Auth/discussions/911.
